src/main/kotlin/org/rust/ide/refactoring/introduceConstant/RsIntroduceConstantHandler.kt

```diff
@@ -54,6 +55,10 @@ private fun RsExpr.isExtractable(): Boolean {
     return when (this) {
         is RsLitExpr -> true
         is RsBinaryExpr -> this.left.isExtractable() && (this.right?.isExtractable() ?: true)
+        is RsPathExpr -> {
+            val target = path.reference?.resolve() as? RsConstant
```

Comment:
Looks like `resolve` here makes things a bit harder because:
- currently, `RsIntroduceConstantHandler#invoke` is called from EDT
- it's quite hard to guarantee that `resolve` is not a long-running operation since its implementation may use indices. Yeah, currently some resolve operations are quite cheap because of new name resolution engine and its precomputation phase but I don't believe that we can track that **all** resolve operations here are fast

So it would be great to extract potential long computation into background here.
I suppose we can introduce something similar to [nonblocking](https://github.com/JetBrains/intellij-community/blob/master/plugins/kotlin/idea/src/org/jetbrains/kotlin/idea/util/nonblocking.kt) function used in kotlin plugin

@vlad20012 @dima74 what do you think?